collect/collect.go

```diff
@@ -559,7 +559,9 @@ func (i *InMemCollector) ProcessSpanImmediately(sp *types.Span, keep bool, sampl
 	if i.Config.GetAddRuleReasonToTrace() {
 		sp.Event.Data["meta.refinery.reason"] = reason
 	}
-
+	if i.Config.GetStressReliefConfig().Mode != "" {
```

Comment:
Can we use the same `if i.hostname !=nil` logic that is used in the `send` function?